pkg/arvo/sys/hoon.hoon

```diff
@@ -9865,6 +9865,8 @@
                   next
                 ?~  u.tyr
                   $(sut q.sut, lon [~ lon], p.heg +(p.heg))
+                ?:  !=(0 p.heg)
+                  $(sut q.sut, lon [~ lon], p.heg (dec p.heg))
```

Comment:
You know, on reading the fond code for myself, I arrived at exactly the same question you had @joemfb: what are we modelling with the nullability of the hoon in an alias (u.tyr here)? The increment to the ^ count is extremely strange. Suppose you have a subject [◊/1 a/2 a/3]. If ◊ is a tune with a mapped to ~, then the wing a resolves to 3. These ~ aliases then seem to be a way to introduce nonlocal skipping behavior in name resolution — why do we want this functionality? In any event, it does seem to be the most consistent to place the ?: after this ?~  u.tyr, as OP has, which answers my remaining question about this code.

Reply:
My earlier guess that this nullability/skip was an old artifact is wrong, it's specifically called out in https://urbit.org/docs/reference/hoon-expressions/advanced/. It's still strange: as far as I can tell, nothing uses this. And the code as written appears to include a double-skip, bypassing the `%face` (replacing `sut` with `q.sut`) *and* incrementing the skip count (`p.heg`).